web/src/lib/components/faces-page/search-people.svelte

```diff
@@ -0,0 +1,88 @@
+<script lang="ts">
+  import SearchBar from '$lib/components/elements/search-bar.svelte';
+  import { maximumLengthSearchPeople, timeBeforeShowLoadingSpinner } from '$lib/constants';
+  import { handleError } from '$lib/utils/handle-error';
+  import { searchNameLocal } from '$lib/utils/person';
+  import { searchPerson, type PersonResponseDto } from '@immich/sdk';
+
+  export let searchName: string;
+  export let searchedPeopleLocal: PersonResponseDto[];
+  export let type: 'searchBar' | 'input';
+  export let numberPeopleToSearch: number = maximumLengthSearchPeople;
+  export let inputClass: string = 'w-full gap-2 bg-immich-bg dark:bg-immich-dark-bg';
+  export let showLoadingSpinner: boolean = false;
+  export let isSearching: boolean = false;
+  export let placeholder: string = 'Name or nickname';
+  export let onReset = () => {};
+  export let onSearch = () => {};
+
+  let searchedPeople: PersonResponseDto[] = [];
+  let searchWord: string;
+  let abortController: AbortController | null = null;
+  let timeout: NodeJS.Timeout | null = null;
+
+  const search = () => {
+    searchedPeopleLocal = searchNameLocal(searchName, searchedPeople, numberPeopleToSearch);
+  };
+
+  export let handleSearch = async (force?: boolean, name?: string) => {
```

Comment:
Why does this method accept a name? I don't see any place where that's passed. Actually, I don't even see point why that method should be exported in the first place

Reply:
I also personally find this method quite hard to read. I'm not 100% sure why that is; I think it's just many ifs for a method I'd expect to be simple :D

Reply:
It's because of
 ```ts
  onMount(async () => {
    const getSearchedPeople = $page.url.searchParams.get(QueryParameter.SEARCHED_PEOPLE);
    if (getSearchedPeople) {
      searchName = getSearchedPeople;
      await handleSearchPeople(true, searchName);
    }
  });
``` 
[here](https://github.com/immich-app/immich/pull/9082/files#diff-206e8cab8ac2af387ea424aad5b67c8669f615a66b8d83eb66271aa81d124998R77)

The search is saved in a query parameter so onMount, if you don't have that `handleSearchPeople`, you don't trigger the search and `searchName` in the `search-people` component is not yet binded to the `searchName` from its parent. 

The issue with that component is that it's doing the search for every components. It was implemented 4 times with different features and bugs, I prefer to have one which is doing everything well.

Reply:
But `handleSearchPeople` is a different method, no? I'm confused...

Reply:
Nope, it's mounted to `handleSearch` like that : 
```svelte
<SearchPeople
  type="searchBar"
  placeholder="Search people"
  onReset={onResetSearchBar}
  onSearch={handleSearch}
  bind:searchName
  bind:searchedPeopleLocal
  bind:isSearching={isSearchingPeople}
  bind:handleSearch={handleSearchPeople}
/>
```
so the parent component can interact with the child method.